### PR TITLE
Dashboard

### PR DESCRIPTION
**Pull request by thomashopkins32**
- Added script to generate the dashboard found at https://github.com/nsls-ii-forge/project-management.
- Added an entry point called `dashboard` that will execute this script with some options.
- There is no integration with the other repository, however. You will need to move the generated file and push it to the `project_management` repo when regenerating the dashboard with new feedstocks.

### CODE REVIEW

File: nsls2forge_utils/cli.py [hunk 1]
```diff
@@ -123,3 +124,21 @@ def all_feedstocks():
         parser.exit(message='Please specify organization and sub-command...\n')
 
     args.func(args)
+
+
+def dashboard():
+    parser = argparse.ArgumentParser(
+        description='Create a dashboard of feedstocks belonging to nsls-ii-forge')
```
[the remark this answers is not in the document]
Comment (thomashopkins32) [reply nsls2forge_utils/cli.py:131]: I thought about that but decided against it due to the difference in naming `nsls-ii-forge` vs `nsls2forge`, unless some other organization is hosting our feedstocks on the same Azure and Anaconda profile.

I guess I didn't see the benefit in adding it since it would only change where the feedstocks are coming from. If you think I should still make the adjustment I will add it.

[the remark this answers is not in the document]
Comment (mrakitin) [reply nsls2forge_utils/cli.py:131]: OK, let's keep it as is for now.

File: .gitignore [hunk 2]
```diff
@@ -90,3 +90,4 @@ feedstocks/*
 coverage.xml
 .coverage.*
 miniconda.sh
+README.md
```
Comment (mrakitin) [inline .gitignore:93]: Why is this change?

Comment (thomashopkins32) [reply .gitignore:93]: Running the `dashboard.py` script in this repo creates that file by default, so I added it to `.gitignore` so it won't overwrite the current `README.rst` and show up on this GitHub repo when it isn't supposed to.

Comment (mrakitin) [reply .gitignore:93]: Ah, got it. Makes sense.

**Comment by thomashopkins32** [comment]
What would I be testing for exactly?

Would it be like:
- Test with no feedstock repos & compare output to expected
- Test with one feedstock repo & compare output to expected

I guess I am a little confused on tests other than these that I would make.

**Comment by mrakitin** [comment]
I think it would make sense to run this tool in CI with all non-archived feedstocks and see that the produced table in the `README.md` file has the expected number of rows (will use `assert` to do the comparison). Also, the tests you mentioned would not hurt (like an empty list of repos and a single repo name passed to the function).

Also, testing the input parameters like the output file name would make sense.

**Comment by thomashopkins32** [comment]
Ok I will work on that. Thanks for the suggestion!

**Comment by thomashopkins32** [comment]
Didn't mean to do the merge commit. Should I merge this PR and do the tests in a separate PR?

**Comment by mrakitin** [comment]
It's fine. It will merge to master cleanly.

**Comment by mrakitin** [comment]
Yeah, maybe tests can be done via a separate PR to avoid infinite growth of this PR :D
